### ISSUE TITLE
mod1, fld1, rem1 broken

### ISSUE THREAD

**Issue by mschauer** [JuliaLang/julia#5570]
```
julia> map(x -> int(mod1(uint(x),uint(5))), 0:15)'
1x16 Array{Int64,2}:
 5  1  2  3  4  5  5  1  2  3  4  5  1  2  3  4
```
It is like this since 0.1.2.

**Comment by mschauer** [comment]
This should perhaps get a bug tag.

**Comment by jiahao** [comment]
Note: `mod1`, `fld1`, `rem1` have no tests.

**Comment by eschnett** [comment]
This should be fixed by #14140: `mod1` was corrected by #10585, `fld1` now has a consistent behaviour for negative values, and `rem1` is deprecated.

**Comment by mschauer** [comment]
Great
